posix_test.go

```diff
@@ -401,6 +401,91 @@ func TestDeleteFile(t *testing.T) {
+		// Path length is > 1024 chars long.
+		{"level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003/object000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001", errFileNameTooLong},
+	}
+
+	// Add path length > 1024 test specially as OS X system does not support 1024 long path.
+	err = errFileNameTooLong
+	if runtime.GOOS != "darwin" {
+		err = nil
+	}
+
+	// Path length is 1024 chars long.
+	testCases = append(testCases, struct {
+		fileName    string
+		expectedErr error
+	}{"level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003/object000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001", err})
+
+	var buf = make([]byte, 5)
+
+	// Create test files for further reading.
+	for _, testCase := range testCases {
+		if err := posix.AppendFile("success-vol", testCase.fileName, []byte("hello, world")); err != testCase.expectedErr {
+			t.Fatalf("case: %s, expected: %s, got: %s", testCase, testCase.expectedErr, err)
+		}
+	}
+
+	for _, testCase := range testCases {
+		if _, err := posix.ReadFile("success-vol", testCase.fileName, 0, buf); err != testCase.expectedErr {
```

Comment:
It'll be good to not to ignore a return value, its good to assert it for expected value.